pkg/apis/tenancy/v1alpha1/types.go

```diff
@@ -0,0 +1,142 @@
+/*
+Copyright 2021 The Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// Workspace describes how clients access (kubelike) APIs
+//
+// +crd
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:subresource:status
+// +kubebuilder:resource:scope=Cluster,categories=kcp
+type Workspace struct {
+	metav1.TypeMeta `json:",inline"`
+	// +optional
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// +optional
+	Spec WorkspaceSpec `json:"spec,omitempty"`
+
+	// +optional
+	Status WorkspaceStatus `json:"status,omitempty"`
+}
+
+// WorkspaceSpec holds the desired state of the Workspace.
+type WorkspaceSpec struct {
+
+	// +optional
```

Comment:
assuming you use kubebuilder's controller-gen, you probably have to use the required+optional markers from kubebuilder, not these from kube. They are ignored.

Reply:
@sttts is it ok to resolve this?

Reply:
if kubebuilder learned that by now, sure.